src/VisualStudio/Core/Def/Implementation/Remote/DefaultRemoteHostClientProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Remote;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Remote
+{
+    internal sealed class DefaultRemoteHostClientProvider : IRemoteHostClientProvider
```

Comment:
Not exported?

Reply:
No, it's provided through a factory, hence why the constructor here uses a different message in the Obsolete attribute.